## Introduction

I would like to use Foursquare to help decide which place in Bangkok, Thailand will be the best to move to if the person was originally from Melbourne, Australia. Both of the countries may seem very different but I would like to explore which places in Bangkok, Thailand have similar facilities as in Melbourne, Australia so that moving to Bangkok from Melbourne will not be too much of a difference. 
    
 

## Data

Factors that will be affecting the results are:
1. Places in Bangkok
2. Places in Melbourne City
3. Common venues in each places (Bangkok & Melbourne City)

The data source that we will be extracting/getting our data from are Foursquare and <a>https://www.geonames.org</a>

Data needed for this analysis:
1. Places and coordinates of those places in Bangkok
2. Places and coordinates of those places in Melbourne City
3. Top 100 places within 1km/1000m of each places using Foursquare(Bangkok & Melbourne City)

##### We will drop columns country and Admin1 which is the state since all of the country and state will be the same. We will also drop column Unnamed:0 since it is irrelevant.

In [2]:
bangkok_data=pd.read_csv('bangkok_data.csv')
melbourne_city_data=pd.read_csv('melbourne_city_data.csv')

In [3]:
bangkok_data.drop('Unnamed: 0',axis=1,inplace=True)
bangkok_data.head()

Place   Code   Country   Admin1  Latitude  Longitude
0               Wattana  10110  Thailand  Bangkok    13.708    100.456
1               Yannawa  10120  Thailand  Bangkok    13.708    100.456
2         Thawi Wattana  10170  Thailand  Bangkok    13.767    100.450
3  Pom Prap Sattru Phai  10100  Thailand  Bangkok    13.733    100.500
4        Samphanthawong  10100  Thailand  Bangkok    13.733    100.500

In [4]:
bangkok_data.drop(['Country','Admin1'],axis=1,inplace=True)
bangkok_data.head()

Place   Code  Latitude  Longitude
0               Wattana  10110    13.708    100.456
1               Yannawa  10120    13.708    100.456
2         Thawi Wattana  10170    13.767    100.450
3  Pom Prap Sattru Phai  10100    13.733    100.500
4        Samphanthawong  10100    13.733    100.500

In [5]:
melbourne_city_data.drop(['Country','Admin2','Unnamed: 0'],axis=1,inplace=True)
melbourne_city_data.head()

Place  Code  Latitude  Longitude
0       Melbourne  3000   -37.813    144.961
1  East Melbourne  3002   -37.813    144.984
2  West Melbourne  3003   -37.809    144.947
3       Melbourne  3004   -37.842    144.976
4       Docklands  3008   -37.818    144.944

Since there is 2 Melbourne I did some research and found out that Postal Code 3004 is indeed also Melbourne however postal code 3004 is also linked to a place called St. Kilda Road Central. To avoid confusion of having 2 Melbournes, I will rename Melbourne with postal code 3004 to St. Kilda Road Central.

In [6]:
melbourne_city_data.at[3,'Place']='St. Kilda Road Central'
melbourne_city_data

Place  Code  Latitude  Longitude
0               Melbourne  3000   -37.813    144.961
1          East Melbourne  3002   -37.813    144.984
2          West Melbourne  3003   -37.809    144.947
3  St. Kilda Road Central  3004   -37.842    144.976
4               Docklands  3008   -37.818    144.944
5         North Melbourne  3051   -37.799    144.949

In [7]:
melbourne_city_data.shape

(6, 4)

In [8]:
bangkok_data.shape

(44, 4)

### Let's visualize the neighbourhoods in Bangkok

##### Let's get Bangkok's coordinates

In [9]:
address = 'Bangkok, TH'

geolocator = Nominatim(user_agent="th_explorer")
location = geolocator.geocode(address)
latitude_bk = location.latitude
longitude_bk = location.longitude
print('The geograpical coordinate of Bangkok are {}, {}.'.format(latitude_bk, longitude_bk))

The geograpical coordinate of Bangkok are 13.7542529, 100.493087.


##### Let's visualize the neighbourhoods

In [10]:
map_bangkok=folium.Map(location=[latitude_bk,longitude_bk],zoom_start=9)

for lat,long,neigh in zip(bangkok_data['Latitude'], bangkok_data['Longitude'], bangkok_data['Place']):
    label=neigh
    label=folium.Popup(label,parse_html=True)
    folium.CircleMarker(
        [lat, long],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_bangkok)
    
map_bangkok

### Let's visualize neighbourhoods in Melbourne City

##### Let's get Melbourne City coordinates

In [11]:
address = 'Melbourne City, AU'

geolocator = Nominatim(user_agent="au_explorer")
location = geolocator.geocode(address)
latitude_mel = location.latitude
longitude_mel = location.longitude
print('The geograpical coordinate of Melbourne City are {}, {}.'.format(latitude_mel, longitude_mel))

The geograpical coordinate of Melbourne City are -37.827558100000005, 144.97985722310904.


##### Lets visualize the neighbourhood in Melbourne City

In [12]:
map_melbourne=folium.Map(location=[latitude_mel,longitude_mel],zoom_start=11)

for lat,long,neigh in zip(melbourne_city_data['Latitude'], melbourne_city_data['Longitude'], melbourne_city_data['Place']):
    label=neigh
    label=folium.Popup(label,parse_html=True)
    folium.CircleMarker(
        [lat, long],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_melbourne)
    
map_melbourne

### Let's use Foursquare to get the top 100 places within 1km of these cities

In [13]:
CLIENT_ID = '24XUTNADDQSXS4ESGZ3YG5GRFMCFZ1NPCPS5X4TRXBWP12UP'
CLIENT_SECRET = 'EFAO0VCYT0X42HXFFPF2ZDWWC3CA4YR0CAFZGUNFPVD0HOVC'
VERSION = '20180605'

In [14]:
import requests

#### Melbourne City

In [15]:
def getNearbyVenues(names, latitudes, longitudes, radius=1000):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            100)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [16]:
melbourne_city_venues = getNearbyVenues(names=melbourne_city_data['Place'],
                                   latitudes=melbourne_city_data['Latitude'],
                                   longitudes=melbourne_city_data['Longitude']
                                  )




Melbourne
East Melbourne
West Melbourne
St. Kilda Road Central
Docklands
North Melbourne


In [17]:
print(melbourne_city_venues.shape)
melbourne_city_venues.head()

(483, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0    Melbourne                -37.813                 144.961   
1    Melbourne                -37.813                 144.961   
2    Melbourne                -37.813                 144.961   
3    Melbourne                -37.813                 144.961   
4    Melbourne                -37.813                 144.961   

                       Venue  Venue Latitude  Venue Longitude  \
0                    Tipo 00      -37.813527       144.961978   
1  Shortstop Coffee & Donuts      -37.811240       144.961000   
2                 Brick Lane      -37.811341       144.959816   
3         Brother Baba Budan      -37.813445       144.962137   
4            Kirk's Wine Bar      -37.813661       144.961351   

       Venue Category  
0  Italian Restaurant  
1          Donut Shop  
2                Café  
3         Coffee Shop  
4            Wine Bar

### Bangkok

In [18]:
bangkok_venues = getNearbyVenues(names=bangkok_data['Place'],
                                   latitudes=bangkok_data['Latitude'],
                                   longitudes=bangkok_data['Longitude']
                                  )

Wattana
Yannawa
Thawi Wattana
Pom Prap Sattru Phai
Samphanthawong
Bang Kholame
Thung Khru
Rat Burana
Bang Bon
Bang Khun Thian
Phasi Charoen
Nong Khaem
Bang Khae
Taling Chan
Phra Nakhon
Lak Si
Sai Mai
Bang Khen
Khanna Yao
Lat Phrao
Saphan Sung B39
Bung Kum
Pra Vet
Suang Luang
Phra Khanong
Dusit
Wangthong Lang
Huai Khwang
Pahtum Wan
Ding Daeng
Bang Rak
Khlong Samwa
Min Buri
Lat Krabang
Nong Chok
Bangkok Yai
Thon Buri
Khlong San
Bang Plat
Bangkok Noi
Bang Su
Bang Kapi
Bang Na
Phaya Thai


In [19]:
print(bangkok_venues.shape)
bangkok_venues.head()

(2417, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0      Wattana                 13.708                 100.456   
1      Wattana                 13.708                 100.456   
2      Wattana                 13.708                 100.456   
3      Wattana                 13.708                 100.456   
4      Wattana                 13.708                 100.456   

                         Venue  Venue Latitude  Venue Longitude  \
0       Starbucks (สตาร์บัคส์)       13.709113       100.452060   
1  ก๋วยเตี๋ยวต้มยำตำลึง นายเอก       13.709675       100.453821   
2  7-Eleven (เซเว่น อีเลฟเว่น)       13.710411       100.455035   
3             sorry i'm hungry       13.710255       100.454076   
4         TORA Yakiniku x Café       13.705987       100.447110   

      Venue Category  
0        Coffee Shop  
1       Noodle House  
2  Convenience Store  
3       Burger Joint  
4          BBQ Joint

## Analysing each neighbourhood

### Melbourne City

In [20]:
melbourne_onehot= pd.get_dummies(melbourne_city_venues[['Venue Category']],prefix="",prefix_sep="")
melbourne_onehot['Neighbourhood']=melbourne_city_venues['Neighborhood']
fixed_columns=[melbourne_onehot.columns[-1]]+list(melbourne_onehot.columns[:-1])
melbourne_onehot=melbourne_onehot[fixed_columns]
melbourne_onehot.head()


Neighbourhood  African Restaurant  Aquarium  Art Gallery  \
0     Melbourne                   0         0            0   
1     Melbourne                   0         0            0   
2     Melbourne                   0         0            0   
3     Melbourne                   0         0            0   
4     Melbourne                   0         0            0   

   Arts & Crafts Store  Arts & Entertainment  Asian Restaurant  \
0                    0                     0                 0   
1                    0                     0                 0   
2                    0                     0                 0   
3                    0                     0                 0   
4                    0                     0                 0   

   Australian Restaurant  Auto Dealership  Bakery  Bar  Beer Garden  Boutique  \
0                      0                0       0    0            0         0   
1                      0                0       0    0            0         0   
2                      0                0       0    0            0         0   
3                      0                0       0    0            0         0   
4                      0                0       0    0            0         0   

   Brazilian Restaurant  Breakfast Spot  Bridge  Bubble Tea Shop  Buffet  \
0                     0               0       0                0       0   
1                     0               0       0                0       0   
2                     0               0       0                0       0   
3                     0               0       0                0       0   
4                     0               0       0                0       0   

   Building  Burger Joint  Bus Stop  Café  Cambodian Restaurant  Candy Store  \
0         0             0         0     0                     0            0   
1         0             0         0     0                     0            0   
2         0             0         0     1                     0            0   
3         0             0         0     0                     0            0   
4         0             0         0     0                     0            0   

   Chinese Restaurant  Chocolate Shop  Clothing Store  Cocktail Bar  \
0                   0               0               0             0   
1                   0               0               0             0   
2                   0               0               0             0   
3                   0               0               0             0   
4                   0               0               0             0   

   Coffee Shop  Comedy Club  Comic Shop  Convenience Store  Cosmetics Shop  \
0            0            0           0                  0               0   
1            0            0           0                  0               0   
2            0            0           0                  0               0   
3            1            0           0                  0               0   
4            0            0           0                  0               0   

   Creperie  Cricket Ground  Dessert Shop  Dim Sum Restaurant  Dog Run  \
0         0               0             0                   0        0   
1         0               0             0                   0        0   
2         0               0             0                   0        0   
3         0               0             0                   0        0   
4         0               0             0                   0        0   

   Donut Shop  Dumpling Restaurant  Fast Food Restaurant  Filipino Restaurant  \
0           0                    0                     0                    0   
1           1                    0                     0                    0   
2           0                    0                     0                    0   
3           0                    0                     0                    0   
4           0                    0                     0     

In [21]:
melbourne_grouped = melbourne_onehot.groupby('Neighbourhood').mean().reset_index()
melbourne_grouped

Neighbourhood  African Restaurant  Aquarium  Art Gallery  \
0               Docklands            0.000000  0.000000     0.000000   
1          East Melbourne            0.000000  0.000000     0.000000   
2               Melbourne            0.000000  0.000000     0.010000   
3         North Melbourne            0.027778  0.027778     0.027778   
4  St. Kilda Road Central            0.000000  0.000000     0.000000   
5          West Melbourne            0.010000  0.000000     0.000000   

   Arts & Crafts Store  Arts & Entertainment  Asian Restaurant  \
0                 0.00                  0.01              0.01   
1                 0.01                  0.00              0.01   
2                 0.00                  0.00              0.01   
3                 0.00                  0.00              0.00   
4                 0.00                  0.00              0.00   
5                 0.00                  0.00              0.01   

   Australian Restaurant  Auto Dealership    Bakery       Bar  Beer Garden  \
0               0.010000             0.01  0.020000  0.030000     0.010000   
1               0.040000             0.00  0.000000  0.060000     0.010000   
2               0.000000             0.00  0.000000  0.080000     0.010000   
3               0.000000             0.00  0.027778  0.055556     0.000000   
4               0.085106             0.00  0.000000  0.021277     0.021277   
5               0.000000             0.00  0.020000  0.040000     0.000000   

   Boutique  Brazilian Restaurant  Breakfast Spot  Bridge  Bubble Tea Shop  \
0      0.00                  0.00        0.000000    0.01             0.00   
1      0.00                  0.00        0.010000    0.00             0.00   
2      0.01                  0.01        0.000000    0.00             0.02   
3      0.00                  0.00        0.000000    0.00             0.00   
4      0.00                  0.00        0.021277    0.00             0.00   
5      0.00                  0.00        0.000000    0.00             0.00   

   Buffet  Building  Burger Joint  Bus Stop      Café  Cambodian Restaurant  \
0    0.01      0.01      0.010000  0.000000  0.140000                  0.01   
1    0.00      0.00      0.010000  0.000000  0.180000                  0.00   
2    0.00      0.00      0.020000  0.000000  0.070000                  0.00   
3    0.00      0.00      0.027778  0.000000  0.333333                  0.00   
4    0.00      0.00      0.000000  0.021277  0.297872                  0.00   
5    0.01      0.00      0.020000  0.000000  0.200000                  0.01   

   Candy Store  Chinese Restaurant  Chocolate Shop  Clothing Store  \
0         0.00                0.01            0.00            0.01   
1         0.00                0.01            0.00            0.01   
2         0.02                0.03            0.01            0.01   
3         0.00                0.00            0.00            0.00   
4         0.00                0.00            0.00            0.00   
5         0.00                0.01            0.00            0.00   

   Cocktail Bar  Coffee Shop  Comedy Club  Comic Shop  Convenience Store  \
0      0.000000     0.060000         0.00        0.00           0.000000   
1      0.040000     0.060000         0.00        0.00           0.000000   
2      0.040000     0.070000         0.00        0.01           0.000000   
3      0.027778     0.027778         0.00        0.00           0.000000   
4      0.000000     0.042553         0.00        0.00           0.021277   
5      0.010000     0.020000         0.01        0.00           0.000000   

   Cosmetics Shop  Creperie  Cricket Ground  Dessert Shop  Dim Sum Restaurant  \
0            0.00      0.00            0.00          0.02                0.00   
1            0.00      0.00            0.04          0.01                0.00   
2            0.02      0.01            0.00          0.03                0.02   
3            0.00      0.00            0.00     

In [22]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

#### Getting top 10 venues of each neighbourhood

In [23]:
top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighbourhood']
for ind in np.arange(top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
melbourne_venues_sorted = pd.DataFrame(columns=columns)
melbourne_venues_sorted['Neighbourhood'] = melbourne_grouped['Neighbourhood']

for ind in np.arange(melbourne_grouped.shape[0]):
    melbourne_venues_sorted.iloc[ind, 1:] = return_most_common_venues(melbourne_grouped.iloc[ind, :], top_venues)

melbourne_venues_sorted.head()

Neighbourhood 1st Most Common Venue  2nd Most Common Venue  \
0               Docklands                  Café            Coffee Shop   
1          East Melbourne                  Café            Coffee Shop   
2               Melbourne                   Bar                   Café   
3         North Melbourne                  Café        Thai Restaurant   
4  St. Kilda Road Central                  Café  Australian Restaurant   

  3rd Most Common Venue  4th Most Common Venue 5th Most Common Venue  \
0                 Hotel     Italian Restaurant                   Bar   
1                   Bar        Thai Restaurant                 Hotel   
2           Coffee Shop      Korean Restaurant          Cocktail Bar   
3                   Bar  Indonesian Restaurant           Pizza Place   
4                 Hotel     Light Rail Station                  Park   

  6th Most Common Venue 7th Most Common Venue  8th Most Common Venue  \
0            Restaurant            Steakhouse                   Park   
1          Cocktail Bar        Cricket Ground  Australian Restaurant   
2         Shopping Mall    Chinese Restaurant           Dessert Shop   
3    Persian Restaurant     Korean Restaurant          Indie Theater   
4           Coffee Shop     French Restaurant  Portuguese Restaurant   

  9th Most Common Venue 10th Most Common Venue  
0   Japanese Restaurant            Supermarket  
1   Japanese Restaurant               Wine Bar  
2    Italian Restaurant        Thai Restaurant  
3                 Hotel                 Hostel  
4                 Plaza            Pizza Place

### Bangkok

In [24]:
bangkok_onehot= pd.get_dummies(bangkok_venues[['Venue Category']],prefix="",prefix_sep="")
bangkok_onehot['Neighbourhood']=bangkok_venues['Neighborhood']
fixed_columns=[bangkok_onehot.columns[-1]]+list(bangkok_onehot.columns[:-1])
bangkok_onehot=bangkok_onehot[fixed_columns]
bangkok_onehot.head()

Neighbourhood  Accessories Store  American Restaurant  Arcade  Art Gallery  \
0       Wattana                  0                    0       0            0   
1       Wattana                  0                    0       0            0   
2       Wattana                  0                    0       0            0   
3       Wattana                  0                    0       0            0   
4       Wattana                  0                    0       0            0   

   Art Museum  Asian Restaurant  Athletics & Sports  BBQ Joint  \
0           0                 0                   0          0   
1           0                 0                   0          0   
2           0                 0                   0          0   
3           0                 0                   0          0   
4           0                 0                   0          1   

   Badminton Court  Bakery  Bar  Beer Bar  Beer Garden  Big Box Store  \
0                0       0    0         0            0              0   
1                0       0    0         0            0              0   
2                0       0    0         0            0              0   
3                0       0    0         0            0              0   
4                0       0    0         0            0              0   

   Bike Trail  Bistro  Boat or Ferry  Bookstore  Boutique  Bowling Alley  \
0           0       0              0          0         0              0   
1           0       0              0          0         0              0   
2           0       0              0          0         0              0   
3           0       0              0          0         0              0   
4           0       0              0          0         0              0   

   Breakfast Spot  Bubble Tea Shop  Buddhist Temple  Buffet  Burger Joint  \
0               0                0                0       0             0   
1               0                0                0       0             0   
2               0                0                0       0             0   
3               0                0                0       0             1   
4               0                0                0       0             0   

   Bus Station  Bus Stop  Cafeteria  Café  Canal  Cantonese Restaurant  \
0            0         0          0     0      0                     0   
1            0         0          0     0      0                     0   
2            0         0          0     0      0                     0   
3            0         0          0     0      0                     0   
4            0         0          0     0      0                     0   

   Chinese Restaurant  Clothing Store  Cocktail Bar  Coffee Shop  \
0                   0               0             0            1   
1                   0               0             0            0   
2                   0               0             0            0   
3                   0               0             0            0   
4                   0               0             0            0   

   College Bookstore  College Cafeteria  Comfort Food Restaurant  Comic Shop  \
0                  0                  0                        0           0   
1                  0                  0                        0           0   
2                  0                  0                        0           0   
3                  0                  0                        0           0   
4                  0                  0                        0           0   

   Convenience Store  Coworking Space  Department Store  Dessert Shop  \
0                  0                0                 0             0   
1                  0                0                 0             0   
2                  1                0                 0             0   
3                  0                0                 0             0   
4                  0                0                 0          

In [25]:
bangkok_grouped=bangkok_onehot.groupby('Neighbourhood').mean().reset_index()
bangkok_grouped.head()

Neighbourhood  Accessories Store  American Restaurant  Arcade  Art Gallery  \
0      Bang Bon                0.0                  0.0     0.0          0.0   
1     Bang Kapi                0.0                  0.0     0.0          0.0   
2     Bang Khae                0.0                  0.0     0.0          0.0   
3     Bang Khen                0.0                  0.0     0.0          0.0   
4  Bang Kholame                0.0                  0.0     0.0          0.0   

   Art Museum  Asian Restaurant  Athletics & Sports  BBQ Joint  \
0         0.0          0.000000            0.000000   0.000000   
1         0.0          0.000000            0.000000   0.020000   
2         0.0          0.090909            0.000000   0.000000   
3         0.0          0.057692            0.000000   0.019231   
4         0.0          0.000000            0.027778   0.027778   

   Badminton Court    Bakery     Bar  Beer Bar  Beer Garden  Big Box Store  \
0              0.0  0.000000  0.0625       0.0     0.000000            0.0   
1              0.0  0.010000  0.0000       0.0     0.000000            0.0   
2              0.0  0.000000  0.0000       0.0     0.000000            0.0   
3              0.0  0.019231  0.0000       0.0     0.019231            0.0   
4              0.0  0.027778  0.0000       0.0     0.000000            0.0   

   Bike Trail  Bistro  Boat or Ferry  Bookstore  Boutique  Bowling Alley  \
0         0.0     0.0            0.0   0.000000       0.0           0.00   
1         0.0     0.0            0.0   0.000000       0.0           0.01   
2         0.0     0.0            0.0   0.090909       0.0           0.00   
3         0.0     0.0            0.0   0.038462       0.0           0.00   
4         0.0     0.0            0.0   0.000000       0.0           0.00   

   Breakfast Spot  Bubble Tea Shop  Buddhist Temple  Buffet  Burger Joint  \
0             0.0              0.0              0.0     0.0      0.000000   
1             0.0              0.0              0.0     0.0      0.000000   
2             0.0              0.0              0.0     0.0      0.000000   
3             0.0              0.0              0.0     0.0      0.019231   
4             0.0              0.0              0.0     0.0      0.027778   

   Bus Station  Bus Stop  Cafeteria      Café  Canal  Cantonese Restaurant  \
0     0.000000       0.0        0.0  0.062500    0.0                   0.0   
1     0.000000       0.0        0.0  0.000000    0.0                   0.0   
2     0.090909       0.0        0.0  0.000000    0.0                   0.0   
3     0.000000       0.0        0.0  0.019231    0.0                   0.0   
4     0.000000       0.0        0.0  0.027778    0.0                   0.0   

   Chinese Restaurant  Clothing Store  Cocktail Bar  Coffee Shop  \
0              0.0625            0.00           0.0     0.000000   
1              0.0100            0.04           0.0     0.100000   
2              0.0000            0.00           0.0     0.000000   
3              0.0000            0.00           0.0     0.019231   
4              0.0000            0.00           0.0     0.111111   

   College Bookstore  College Cafeteria  Comfort Food Restaurant  Comic Shop  \
0                0.0                0.0                      0.0        0.00   
1                0.0                0.0                      0.0        0.01   
2                0.0                0.0                      0.0        0.00   
3                0.0                0.0                      0.0        0.00   
4                0.0                0.0                      0.0        0.00   

   Convenience Store  Coworking Space  Department Store  Dessert Shop  \
0           0.062500              0.0          0.062500      0.000000   
1           0.020000              0.0          0.020000      0.030000   
2           0.181818              0.0          0.090909      0.090909   
3           0.096154              0.0          0.019231      0.019231   
4          

#### Getting top 10 venues of each neighbourhood

In [26]:
top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighbourhood']
for ind in np.arange(top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
bangkok_venues_sorted = pd.DataFrame(columns=columns)
bangkok_venues_sorted['Neighbourhood'] = bangkok_grouped['Neighbourhood']

for ind in np.arange(melbourne_grouped.shape[0]):
    bangkok_venues_sorted.iloc[ind, 1:] = return_most_common_venues(bangkok_grouped.iloc[ind, :], top_venues)

bangkok_venues_sorted.head()

Neighbourhood 1st Most Common Venue 2nd Most Common Venue  \
0      Bang Bon          Noodle House         Grocery Store   
1     Bang Kapi           Coffee Shop       Thai Restaurant   
2     Bang Khae     Convenience Store                Market   
3     Bang Khen     Convenience Store          Noodle House   
4  Bang Kholame          Noodle House           Coffee Shop   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0          Intersection                  Café    Som Tum Restaurant   
1   Japanese Restaurant          Noodle House  Fast Food Restaurant   
2          Noodle House             Bookstore          Tennis Court   
3      Asian Restaurant             Bookstore      Ramen Restaurant   
4       Thai Restaurant            Food Truck  Gym / Fitness Center   

  6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
0      Department Store           Snack Place     Convenience Store   
1        Clothing Store          Dessert Shop     Hotpot Restaurant   
2  Fast Food Restaurant           Bus Station          Dessert Shop   
3        Ice Cream Shop     Hotpot Restaurant            Steakhouse   
4     Convenience Store                  Café         Shopping Mall   

   9th Most Common Venue 10th Most Common Venue  
0     Chinese Restaurant                    Bar  
1         Ice Cream Shop             Steakhouse  
2       Department Store       Asian Restaurant  
3  Vietnamese Restaurant          Shopping Mall  
4           Burger Joint     Som Tum Restaurant

## Now lets combine the Melbourne City data with the Bangkok data to cluster them

In [27]:
combined_venues=pd.concat([melbourne_city_venues,bangkok_venues],ignore_index=True)

In [28]:
combined_venues.shape

(2900, 7)

In [29]:
combined_onehot= pd.get_dummies(combined_venues[['Venue Category']],prefix="",prefix_sep="")
combined_onehot['Neighbourhood']=combined_venues['Neighborhood']
fixed_columns=[combined_onehot.columns[-1]]+list(combined_onehot.columns[:-1])
combined_onehot=combined_onehot[fixed_columns]


In [30]:
combined_grouped = combined_onehot.groupby('Neighbourhood').mean().reset_index()
combined_grouped.head()

Neighbourhood  Accessories Store  African Restaurant  American Restaurant  \
0      Bang Bon                0.0                 0.0                  0.0   
1     Bang Kapi                0.0                 0.0                  0.0   
2     Bang Khae                0.0                 0.0                  0.0   
3     Bang Khen                0.0                 0.0                  0.0   
4  Bang Kholame                0.0                 0.0                  0.0   

   Aquarium  Arcade  Art Gallery  Art Museum  Arts & Crafts Store  \
0       0.0     0.0          0.0         0.0                  0.0   
1       0.0     0.0          0.0         0.0                  0.0   
2       0.0     0.0          0.0         0.0                  0.0   
3       0.0     0.0          0.0         0.0                  0.0   
4       0.0     0.0          0.0         0.0                  0.0   

   Arts & Entertainment  Asian Restaurant  Athletics & Sports  \
0                   0.0          0.000000            0.000000   
1                   0.0          0.000000            0.000000   
2                   0.0          0.090909            0.000000   
3                   0.0          0.057692            0.000000   
4                   0.0          0.000000            0.027778   

   Australian Restaurant  Auto Dealership  BBQ Joint  Badminton Court  \
0                    0.0              0.0   0.000000              0.0   
1                    0.0              0.0   0.020000              0.0   
2                    0.0              0.0   0.000000              0.0   
3                    0.0              0.0   0.019231              0.0   
4                    0.0              0.0   0.027778              0.0   

     Bakery     Bar  Beer Bar  Beer Garden  Big Box Store  Bike Trail  Bistro  \
0  0.000000  0.0625       0.0     0.000000            0.0         0.0     0.0   
1  0.010000  0.0000       0.0     0.000000            0.0         0.0     0.0   
2  0.000000  0.0000       0.0     0.000000            0.0         0.0     0.0   
3  0.019231  0.0000       0.0     0.019231            0.0         0.0     0.0   
4  0.027778  0.0000       0.0     0.000000            0.0         0.0     0.0   

   Boat or Ferry  Bookstore  Boutique  Bowling Alley  Brazilian Restaurant  \
0            0.0   0.000000       0.0           0.00                   0.0   
1            0.0   0.000000       0.0           0.01                   0.0   
2            0.0   0.090909       0.0           0.00                   0.0   
3            0.0   0.038462       0.0           0.00                   0.0   
4            0.0   0.000000       0.0           0.00                   0.0   

   Breakfast Spot  Bridge  Bubble Tea Shop  Buddhist Temple  Buffet  Building  \
0             0.0     0.0              0.0              0.0     0.0       0.0   
1             0.0     0.0              0.0              0.0     0.0       0.0   
2             0.0     0.0              0.0              0.0     0.0       0.0   
3             0.0     0.0              0.0              0.0     0.0       0.0   
4             0.0     0.0              0.0              0.0     0.0       0.0   

   Burger Joint  Bus Station  Bus Stop  Cafeteria      Café  \
0      0.000000     0.000000       0.0        0.0  0.062500   
1      0.000000     0.000000       0.0        0.0  0.000000   
2      0.000000     0.090909       0.0        0.0  0.000000   
3      0.019231     0.000000       0.0        0.0  0.019231   
4      0.027778     0.000000       0.0        0.0  0.027778   

   Cambodian Restaurant  Canal  Candy Store  Cantonese Restaurant  \
0                   0.0    0.0          0.0                   0.0   
1                   0.0    0.0          0.0                   0.0   
2                   0.0    0.0          0.0                   0.0   
3                   0.0    0.0          0.0                   0.0   
4                   0.0    0.0          0.0                   0.0   

   Chinese Restaurant  Chocolate Shop  Clothi

In [31]:
top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighbourhood']
for ind in np.arange(top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
combined_sorted = pd.DataFrame(columns=columns)
combined_sorted['Neighbourhood'] = combined_grouped['Neighbourhood']

for ind in np.arange(combined_grouped.shape[0]):
    combined_sorted.iloc[ind, 1:] = return_most_common_venues(combined_grouped.iloc[ind, :], top_venues)

combined_sorted.head()

Neighbourhood 1st Most Common Venue 2nd Most Common Venue  \
0      Bang Bon          Noodle House         Grocery Store   
1     Bang Kapi           Coffee Shop       Thai Restaurant   
2     Bang Khae     Convenience Store      Department Store   
3     Bang Khen     Convenience Store          Noodle House   
4  Bang Kholame          Noodle House           Coffee Shop   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0                  Café                   Bar    Chinese Restaurant   
1          Noodle House  Fast Food Restaurant   Japanese Restaurant   
2          Dessert Shop                Market           Bus Station   
3      Asian Restaurant      Ramen Restaurant             Bookstore   
4       Thai Restaurant  Gym / Fitness Center            Food Truck   

  6th Most Common Venue 7th Most Common Venue  8th Most Common Venue  \
0           Snack Place     Convenience Store     Som Tum Restaurant   
1        Clothing Store      Ramen Restaurant           Dessert Shop   
2          Tennis Court          Noodle House   Fast Food Restaurant   
3     Hotpot Restaurant        Ice Cream Shop  Vietnamese Restaurant   
4         Shopping Mall                Market            Supermarket   

    9th Most Common Venue 10th Most Common Venue  
0        Department Store           Intersection  
1  Shabu-Shabu Restaurant            Flea Market  
2        Asian Restaurant              Bookstore  
3              Steakhouse                Stadium  
4             Gas Station                   Café

## Clustering Data

In [77]:
# set number of clusters
kclusters = 5

combined_grouped_clustering = combined_grouped.drop('Neighbourhood', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(combined_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:10] 

array([0, 3, 1, 1, 0, 0, 3, 0, 3, 0], dtype=int32)

In [79]:
combined_data=pd.concat([melbourne_city_data,bangkok_data],ignore_index=True)

In [80]:
# add clustering labels
combined_sorted.insert(0, 'Cluster Labels', kmeans.labels_)

combined_merged = combined_data


combined_merged = combined_merged.join(combined_sorted.set_index('Neighbourhood'), on='Place')

combined_merged.head()

Place  Code  Latitude  Longitude  Cluster Labels  \
0               Melbourne  3000   -37.813    144.961             3.0   
1          East Melbourne  3002   -37.813    144.984             2.0   
2          West Melbourne  3003   -37.809    144.947             2.0   
3  St. Kilda Road Central  3004   -37.842    144.976             2.0   
4               Docklands  3008   -37.818    144.944             2.0   

  1st Most Common Venue  2nd Most Common Venue 3rd Most Common Venue  \
0                   Bar                   Café           Coffee Shop   
1                  Café            Coffee Shop                   Bar   
2                  Café      Korean Restaurant    Italian Restaurant   
3                  Café  Australian Restaurant                 Hotel   
4                  Café            Coffee Shop                 Hotel   

  4th Most Common Venue  5th Most Common Venue 6th Most Common Venue  \
0     Korean Restaurant           Cocktail Bar         Shopping Mall   
1       Thai Restaurant  Australian Restaurant        Cricket Ground   
2                   Bar                    Pub     Indian Restaurant   
3    Light Rail Station      French Restaurant           Coffee Shop   
4            Restaurant             Steakhouse                   Bar   

  7th Most Common Venue 8th Most Common Venue 9th Most Common Venue  \
0          Dessert Shop    Italian Restaurant    Chinese Restaurant   
1          Cocktail Bar                 Hotel   Japanese Restaurant   
2          Burger Joint           Pizza Place      Malay Restaurant   
3                  Park           Golf Course       Nature Preserve   
4    Italian Restaurant   Japanese Restaurant                  Park   

  10th Most Common Venue  
0        Thai Restaurant  
1               Wine Bar  
2                  Hotel  
3      Food & Drink Shop  
4            Supermarket

In [78]:
combined_merged.drop('Cluster Labels',axis=1,inplace=True)
combined_sorted.drop('Cluster Labels',axis=1,inplace=True)

In [81]:
combined_merged.dropna(subset=['Cluster Labels'],axis=0,inplace=True)

In [82]:
print(combined_merged.shape)


(47, 15)


In [83]:
combined_merged[['Cluster Labels']]=combined_merged[['Cluster Labels']].astype('int32')

### Lets see the clusters in the map

#### Australia

In [84]:
# create map
map_clusters = folium.Map(location=[latitude_mel, longitude_mel], zoom_start=12)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(combined_merged['Latitude'], combined_merged['Longitude'], combined_merged['Place'], combined_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

#### Bangkok

In [85]:
# create map
map_clusters = folium.Map(location=[latitude_bk, longitude_bk], zoom_start=9)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(combined_merged['Latitude'], combined_merged['Longitude'], combined_merged['Place'], combined_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

## Analysing each cluster

### Cluster 1

In [86]:
combined_merged.loc[combined_merged['Cluster Labels'] == 0, combined_merged.columns[[0] + list(range(5, combined_merged.shape[1]))]]

Place 1st Most Common Venue 2nd Most Common Venue  \
6                Wattana          Noodle House           Coffee Shop   
7                Yannawa          Noodle House           Coffee Shop   
9   Pom Prap Sattru Phai          Noodle House     Convenience Store   
10        Samphanthawong          Noodle House     Convenience Store   
11          Bang Kholame          Noodle House           Coffee Shop   
14              Bang Bon          Noodle House         Grocery Store   
15       Bang Khun Thian          Noodle House         Grocery Store   
17            Nong Khaem       Thai Restaurant          Noodle House   
20           Phra Nakhon          Noodle House      Asian Restaurant   
31                 Dusit          Noodle House       Thai Restaurant   
32        Wangthong Lang          Noodle House       Thai Restaurant   
33           Huai Khwang          Noodle House       Thai Restaurant   
34            Pahtum Wan          Noodle House       Thai Restaurant   
43            Khlong San          Noodle House     Convenience Store   
44             Bang Plat       Thai Restaurant          Noodle House   
45           Bangkok Noi       Thai Restaurant          Noodle House   
46               Bang Su          Noodle House     Convenience Store   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
6        Thai Restaurant  Gym / Fitness Center            Food Truck   
7        Thai Restaurant  Gym / Fitness Center            Food Truck   
9       Asian Restaurant    Seafood Restaurant    Chinese Restaurant   
10      Asian Restaurant    Seafood Restaurant    Chinese Restaurant   
11       Thai Restaurant  Gym / Fitness Center            Food Truck   
14                  Café                   Bar    Chinese Restaurant   
15                  Café                   Bar    Chinese Restaurant   
17           Flea Market     Convenience Store       Badminton Court   
20     Convenience Store    Chinese Restaurant           Coffee Shop   
31           Coffee Shop     Convenience Store      Asian Restaurant   
32           Coffee Shop     Convenience Store      Asian Restaurant   
33           Coffee Shop     Convenience Store      Asian Restaurant   
34           Coffee Shop     Convenience Store      Asian Restaurant   
43      Asian Restaurant    Seafood Restaurant    Chinese Restaurant   
44           Coffee Shop     Convenience Store          Dessert Shop   
45           Coffee Shop     Convenience Store          Dessert Shop   
46       Thai Restaurant           Coffee Shop        Ice Cream Shop   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
6          Shopping Mall                Market           Supermarket   
7          Shopping Mall                Market           Supermarket   
9        Thai Restaurant          Dessert Shop                  Café   
10       Thai Restaurant          Dessert Shop                  Café   
11         Shopping Mall                Market           Supermarket   
14           Snack Place     Convenience Store    Som Tum Restaurant   
15           Snack Place     Convenience Store    Som Tum Restaurant   
17      Toy / Game Store                  Café            Steakhouse   
20     Hotpot Restaurant          Burger Joint    Seafood Restaurant   
31                  Café    Som Tum Restaurant    Chinese Restaurant   
32                  Café    Som Tum Restaurant    Chinese Restaurant   
33                  Café    Som Tum Restaurant    Chinese Restaurant   
34                  Café    Som Tum Restaurant    Chinese Restaurant   
43       Thai Restaurant          Dessert Shop                  Café   
44                  Café                Bakery                Palace   
45                  Café                Bakery                Palace   
46         Train Station             BBQ Joint            Restaurant   

   9th Most Common Venue 10th Most Common Venue  
6            Gas Station                   Café  
7            Gas Station    

#### Cluster 2

In [87]:
combined_merged.loc[combined_merged['Cluster Labels'] == 1, combined_merged.columns[[0] + list(range(5, combined_merged.shape[1]))]]

Place 1st Most Common Venue 2nd Most Common Venue  \
18     Bang Khae     Convenience Store      Department Store   
22       Sai Mai     Convenience Store          Noodle House   
23     Bang Khen     Convenience Store          Noodle House   
28       Pra Vet            Food Truck     Convenience Store   
29   Suang Luang            Food Truck     Convenience Store   
30  Phra Khanong     Convenience Store          Noodle House   
40     Nong Chok  Fast Food Restaurant      Asian Restaurant   
41   Bangkok Yai     Convenience Store          Noodle House   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
18          Dessert Shop                Market           Bus Station   
22      Asian Restaurant      Ramen Restaurant             Bookstore   
23      Asian Restaurant      Ramen Restaurant             Bookstore   
28      Asian Restaurant           Flea Market          Noodle House   
29      Asian Restaurant           Flea Market          Noodle House   
30           Coffee Shop                 Hotel  Fast Food Restaurant   
40            Food Truck           Supermarket           Flea Market   
41                  Food    Seafood Restaurant              Pharmacy   

   6th Most Common Venue 7th Most Common Venue  8th Most Common Venue  \
18          Tennis Court          Noodle House   Fast Food Restaurant   
22     Hotpot Restaurant        Ice Cream Shop  Vietnamese Restaurant   
23     Hotpot Restaurant        Ice Cream Shop  Vietnamese Restaurant   
28           Coffee Shop    Som Tum Restaurant          Boat or Ferry   
29           Coffee Shop    Som Tum Restaurant          Boat or Ferry   
30         Train Station    Italian Restaurant        Thai Restaurant   
40       Thai Restaurant                  Park          Shopping Mall   
41                Market              Beer Bar      Food & Drink Shop   

   9th Most Common Venue 10th Most Common Venue  
18      Asian Restaurant              Bookstore  
22            Steakhouse                Stadium  
23            Steakhouse                Stadium  
28                  Pier                   Food  
29                  Pier                   Food  
30        Ice Cream Shop     Chinese Restaurant  
40  Gym / Fitness Center    Fried Chicken Joint  
41            Food Court            Flower Shop

#### Cluster 3

In [88]:
combined_merged.loc[combined_merged['Cluster Labels'] == 2, combined_merged.columns[[0] + list(range(5, combined_merged.shape[1]))]]

Place 1st Most Common Venue  2nd Most Common Venue  \
1          East Melbourne                  Café            Coffee Shop   
2          West Melbourne                  Café      Korean Restaurant   
3  St. Kilda Road Central                  Café  Australian Restaurant   
4               Docklands                  Café            Coffee Shop   
5         North Melbourne                  Café        Thai Restaurant   

  3rd Most Common Venue 4th Most Common Venue  5th Most Common Venue  \
1                   Bar       Thai Restaurant  Australian Restaurant   
2    Italian Restaurant                   Bar                    Pub   
3                 Hotel    Light Rail Station      French Restaurant   
4                 Hotel            Restaurant             Steakhouse   
5                   Bar    Persian Restaurant           Burger Joint   

  6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
1        Cricket Ground          Cocktail Bar                 Hotel   
2     Indian Restaurant          Burger Joint           Pizza Place   
3           Coffee Shop                  Park           Golf Course   
4                   Bar    Italian Restaurant   Japanese Restaurant   
5               Dog Run          Cocktail Bar           Coffee Shop   

  9th Most Common Venue 10th Most Common Venue  
1   Japanese Restaurant               Wine Bar  
2      Malay Restaurant                  Hotel  
3       Nature Preserve      Food & Drink Shop  
4                  Park            Supermarket  
5            Taco Place                    Gym

#### Cluster 4

In [89]:
combined_merged.loc[combined_merged['Cluster Labels'] == 3, combined_merged.columns[[0] + list(range(5, combined_merged.shape[1]))]]

Place 1st Most Common Venue 2nd Most Common Venue  \
0         Melbourne                   Bar                  Café   
8     Thawi Wattana           Coffee Shop       Thai Restaurant   
12       Thung Khru           Coffee Shop       Thai Restaurant   
13       Rat Burana           Coffee Shop       Thai Restaurant   
16    Phasi Charoen           Coffee Shop   Japanese Restaurant   
19      Taling Chan           Coffee Shop       Thai Restaurant   
26  Saphan Sung B39           Coffee Shop       Thai Restaurant   
27         Bung Kum           Coffee Shop       Thai Restaurant   
35       Ding Daeng     Convenience Store          Noodle House   
36         Bang Rak           Coffee Shop   Japanese Restaurant   
39      Lat Krabang       Thai Restaurant     Convenience Store   
42        Thon Buri          Noodle House     Convenience Store   
47        Bang Kapi           Coffee Shop       Thai Restaurant   
48          Bang Na       Thai Restaurant                  Pool   
49       Phaya Thai     Convenience Store          Noodle House   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0            Coffee Shop     Korean Restaurant          Cocktail Bar   
8           Noodle House   Japanese Restaurant    Seafood Restaurant   
12     Convenience Store    Chinese Restaurant     Hotpot Restaurant   
13     Convenience Store    Chinese Restaurant     Hotpot Restaurant   
16             BBQ Joint     Hotpot Restaurant  Fast Food Restaurant   
19          Noodle House   Japanese Restaurant    Seafood Restaurant   
26          Noodle House  Fast Food Restaurant   Japanese Restaurant   
27          Noodle House  Fast Food Restaurant   Japanese Restaurant   
35                  Café           Coffee Shop            Steakhouse   
36                 Hotel      Ramen Restaurant       Thai Restaurant   
39           Coffee Shop          Noodle House                  Café   
42       Thai Restaurant   Japanese Restaurant      Asian Restaurant   
47          Noodle House  Fast Food Restaurant   Japanese Restaurant   
48     Convenience Store                  Food                  Pier   
49                  Café           Coffee Shop            Steakhouse   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
0          Shopping Mall          Dessert Shop    Italian Restaurant   
8      Convenience Store                  Café          Dessert Shop   
12          Noodle House    Seafood Restaurant             BBQ Joint   
13          Noodle House    Seafood Restaurant             BBQ Joint   
16                   Pub      Asian Restaurant           Pizza Place   
19     Convenience Store                  Café          Dessert Shop   
26        Clothing Store      Ramen Restaurant          Dessert Shop   
27        Clothing Store      Ramen Restaurant          Dessert Shop   
35       Thai Restaurant            Restaurant            Food Court   
36                  Café               Gay Bar             BBQ Joint   
39                 Hotel  Gym / Fitness Center            Bike Trail   
42            Steakhouse     Hotpot Restaurant           Coffee Shop   
47        Clothing Store      Ramen Restaurant          Dessert Shop   
48           Flea Market                  Café          Soccer Field   
49       Thai Restaurant            Restaurant            Food Court   

     9th Most Common Venue 10th Most Common Venue  
0       Chinese Restaurant        Thai Restaurant  
8   Shabu-Shabu Restaurant                 Market  
12                     Pub   Fast Food Restaurant  
13                     Pub   Fast Food Restaurant  
16           Shopping Mall      Convenience Store  
19  Shabu-Shabu Restaurant                 Market  
26  Shabu-Shabu Restaurant            Flea Market  
27  Shabu-Shabu Restaurant            Flea Market  
35                  Hostel     Som Tum Restaurant  
36              Restaurant     Chinese Restaurant  
39  Shabu-Shabu Restaurant             Soup Place  
42         

#### Cluster 5

In [90]:
combined_merged.loc[combined_merged['Cluster Labels'] == 4, combined_merged.columns[[0] + list(range(5, combined_merged.shape[1]))]]

Place 1st Most Common Venue 2nd Most Common Venue  \
37  Khlong Samwa          Intersection       Thai Restaurant   
38      Min Buri          Intersection       Thai Restaurant   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
37            Restaurant      Department Store           Coffee Shop   
38            Restaurant      Department Store           Coffee Shop   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
37           Bus Station          Noodle House                  Park   
38           Bus Station          Noodle House                  Park   

   9th Most Common Venue 10th Most Common Venue  
37                  Farm        Floating Market  
38                  Farm        Floating Market

We can see that <b>Cluster 3</b> consist of almost all the places in Melbourne City. <b>Cluster 4</b> contains various places in Bangkok, Thailand and also Melbourne, Melbourne City. <b>Cluster 1, 2 and 5</b> consist of all places in Bangkok, Thailand. 

#### Since there is only Melbourne in Melbourne City that are relatively similar to neighbourhoods/places in Bangkok. I will cluster the neighbourhoods/places in Melbourne City to see which other neighbourhoods/places are relatively similar to Melbourne, Melbourne City.

In [91]:
# set number of clusters
kclusters = 2

melbourne_city_clustering = melbourne_grouped.drop('Neighbourhood', 1)

# run k-means clustering
kmeans_mel = KMeans(n_clusters=kclusters, random_state=0).fit(melbourne_city_clustering)

# check cluster labels generated for each row in the dataframe
kmeans_mel.labels_[0:10] 

array([1, 1, 1, 0, 0, 1], dtype=int32)

In [92]:
melbourne_venues_sorted.insert(0, 'Cluster Labels', kmeans_mel.labels_)

melbourne_city_merged = melbourne_city_data


melbourne_city_merged = melbourne_city_merged.join(melbourne_venues_sorted.set_index('Neighbourhood'), on='Place')

melbourne_city_merged.head() 

Place  Code  Latitude  Longitude  Cluster Labels  \
0               Melbourne  3000   -37.813    144.961               1   
1          East Melbourne  3002   -37.813    144.984               1   
2          West Melbourne  3003   -37.809    144.947               1   
3  St. Kilda Road Central  3004   -37.842    144.976               0   
4               Docklands  3008   -37.818    144.944               1   

  1st Most Common Venue  2nd Most Common Venue 3rd Most Common Venue  \
0                   Bar                   Café           Coffee Shop   
1                  Café            Coffee Shop                   Bar   
2                  Café      Korean Restaurant    Italian Restaurant   
3                  Café  Australian Restaurant                 Hotel   
4                  Café            Coffee Shop                 Hotel   

  4th Most Common Venue 5th Most Common Venue 6th Most Common Venue  \
0     Korean Restaurant          Cocktail Bar         Shopping Mall   
1       Thai Restaurant                 Hotel          Cocktail Bar   
2                   Bar                   Pub     Indian Restaurant   
3    Light Rail Station                  Park           Coffee Shop   
4    Italian Restaurant                   Bar            Restaurant   

  7th Most Common Venue  8th Most Common Venue      9th Most Common Venue  \
0    Chinese Restaurant           Dessert Shop         Italian Restaurant   
1        Cricket Ground  Australian Restaurant        Japanese Restaurant   
2          Burger Joint                 Bakery  Middle Eastern Restaurant   
3     French Restaurant  Portuguese Restaurant                      Plaza   
4            Steakhouse                   Park        Japanese Restaurant   

  10th Most Common Venue  
0        Thai Restaurant  
1               Wine Bar  
2           Dessert Shop  
3            Pizza Place  
4            Supermarket

In [93]:
# create map
mel_map_clusters = folium.Map(location=[latitude_mel, longitude_mel], zoom_start=12)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(melbourne_city_merged['Latitude'], melbourne_city_merged['Longitude'], melbourne_city_merged['Place'], melbourne_city_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(mel_map_clusters)
       
mel_map_clusters

When we run k-means algorithm with 2 clusters we can see that <b>Docklands, East Melbourne, West Melbourne</b> are also similar to<b> Melbourne</b>. Hence, we can say that these areas are also relatively similar to <b> Cluster 2</b> in Bangkok, Thailand.

### Now let's see these places in Melbourne City that are relatively similar to the places in Bangkok, Thailand.

#### Melbourne City

In [94]:
mel=melbourne_city_merged.loc[melbourne_city_merged['Cluster Labels'] == 1, melbourne_city_merged.columns[[0] + list(range(5, melbourne_city_merged.shape[1]))]]
print(mel.shape[0])
mel

4


Place 1st Most Common Venue 2nd Most Common Venue  \
0       Melbourne                   Bar                  Café   
1  East Melbourne                  Café           Coffee Shop   
2  West Melbourne                  Café     Korean Restaurant   
4       Docklands                  Café           Coffee Shop   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
0           Coffee Shop     Korean Restaurant          Cocktail Bar   
1                   Bar       Thai Restaurant                 Hotel   
2    Italian Restaurant                   Bar                   Pub   
4                 Hotel    Italian Restaurant                   Bar   

  6th Most Common Venue 7th Most Common Venue  8th Most Common Venue  \
0         Shopping Mall    Chinese Restaurant           Dessert Shop   
1          Cocktail Bar        Cricket Ground  Australian Restaurant   
2     Indian Restaurant          Burger Joint                 Bakery   
4            Restaurant            Steakhouse                   Park   

       9th Most Common Venue 10th Most Common Venue  
0         Italian Restaurant        Thai Restaurant  
1        Japanese Restaurant               Wine Bar  
2  Middle Eastern Restaurant           Dessert Shop  
4        Japanese Restaurant            Supermarket

#### Bangkok (Cluster 2)

In [95]:
bangkok=combined_merged.loc[combined_merged['Cluster Labels'] == 3, combined_merged.columns[[0] + list(range(5, combined_merged.shape[1]))]]

In [96]:
bangkok.drop(0,axis=0,inplace=True)

In [97]:
print(bangkok.shape[0])
bangkok

14


Place 1st Most Common Venue 2nd Most Common Venue  \
8     Thawi Wattana           Coffee Shop       Thai Restaurant   
12       Thung Khru           Coffee Shop       Thai Restaurant   
13       Rat Burana           Coffee Shop       Thai Restaurant   
16    Phasi Charoen           Coffee Shop   Japanese Restaurant   
19      Taling Chan           Coffee Shop       Thai Restaurant   
26  Saphan Sung B39           Coffee Shop       Thai Restaurant   
27         Bung Kum           Coffee Shop       Thai Restaurant   
35       Ding Daeng     Convenience Store          Noodle House   
36         Bang Rak           Coffee Shop   Japanese Restaurant   
39      Lat Krabang       Thai Restaurant     Convenience Store   
42        Thon Buri          Noodle House     Convenience Store   
47        Bang Kapi           Coffee Shop       Thai Restaurant   
48          Bang Na       Thai Restaurant                  Pool   
49       Phaya Thai     Convenience Store          Noodle House   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
8           Noodle House   Japanese Restaurant    Seafood Restaurant   
12     Convenience Store    Chinese Restaurant     Hotpot Restaurant   
13     Convenience Store    Chinese Restaurant     Hotpot Restaurant   
16             BBQ Joint     Hotpot Restaurant  Fast Food Restaurant   
19          Noodle House   Japanese Restaurant    Seafood Restaurant   
26          Noodle House  Fast Food Restaurant   Japanese Restaurant   
27          Noodle House  Fast Food Restaurant   Japanese Restaurant   
35                  Café           Coffee Shop            Steakhouse   
36                 Hotel      Ramen Restaurant       Thai Restaurant   
39           Coffee Shop          Noodle House                  Café   
42       Thai Restaurant   Japanese Restaurant      Asian Restaurant   
47          Noodle House  Fast Food Restaurant   Japanese Restaurant   
48     Convenience Store                  Food                  Pier   
49                  Café           Coffee Shop            Steakhouse   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
8      Convenience Store                  Café          Dessert Shop   
12          Noodle House    Seafood Restaurant             BBQ Joint   
13          Noodle House    Seafood Restaurant             BBQ Joint   
16                   Pub      Asian Restaurant           Pizza Place   
19     Convenience Store                  Café          Dessert Shop   
26        Clothing Store      Ramen Restaurant          Dessert Shop   
27        Clothing Store      Ramen Restaurant          Dessert Shop   
35       Thai Restaurant            Restaurant            Food Court   
36                  Café               Gay Bar             BBQ Joint   
39                 Hotel  Gym / Fitness Center            Bike Trail   
42            Steakhouse     Hotpot Restaurant           Coffee Shop   
47        Clothing Store      Ramen Restaurant          Dessert Shop   
48           Flea Market                  Café          Soccer Field   
49       Thai Restaurant            Restaurant            Food Court   

     9th Most Common Venue 10th Most Common Venue  
8   Shabu-Shabu Restaurant                 Market  
12                     Pub   Fast Food Restaurant  
13                     Pub   Fast Food Restaurant  
16           Shopping Mall      Convenience Store  
19  Shabu-Shabu Restaurant                 Market  
26  Shabu-Shabu Restaurant            Flea Market  
27  Shabu-Shabu Restaurant            Flea Market  
35                  Hostel     Som Tum Restaurant  
36              Restaurant     Chinese Restaurant  
39  Shabu-Shabu Restaurant             Soup Place  
42           Train Station   Fast Food Restaurant  
47  Shabu-Shabu Restaurant            Flea Market  
48         Floating Market     Yoshoku Restaurant  
49                  Hostel     Som Tum Restaurant

## Results

Our analysis shows that only some places in Melbourne City that are similar to places in Bangkok, Thailand. Most of the places in Melbourne City are quite dissimilar to the places in Bangkok, Thailand. In our analysis we focus on venues that are 1km within each of the places. By using the data we got using Foursquare and k-means, we grouped the venues of all the places in Bangkok and Melbourne City into 5 clusters to find the similar places. Most of the clusters have only places in Melbourne City or places in Bangkok but not both. Only 1 cluster contains a mixture of both. Only Melbourne of Melbourne City was similar to some places in Bangkok. There are 14 places in Bangkok that are similar to places in Melbourne, Melbourne City.

Since there is only 1 places (Melbourne) that are similar to places in Bangkok, we ran k-means on Melbourne City data to find out places that are relatively similar to Melbourne. As a result, we found 3 other places that are similar to Melbourne, Melbourne City. Hence, we can say that these places are relatively similar to the places in Bangkok that are similar to Melbourne, Melbourne City.

## Discussion

In the result, we can observe that only Melbourne, Melbourne City was the most similar to some places in Bangkok. This maybe because the top 10 venues in Melbourne, Melbourne City has the most Asian restaurants compared to the other places in Melbourne City. There are 14 places in Bangkok that are similar to Melbourne, Melbourne City. Hence, we can say that it would be better for people that are thinking to move from Melbourne, Melbourne City to Bangkok to move to these 14 places in Bangkok due to the similarity of venues between these places.

However, we also found 3 other places in Melbourne City that are relatively similar to Melbourne, Melbourne City. Even though it might not be too similar to where they are coming from from Melbourne City it is still relatively similar or similar enough for them to not feel too foreign when moving to these 14 places in Bangkok. 

## Conclusion

The purpose of this analysis is to discover similar places between Melbourne City, Australia and Bangkok, Thailand so that people that are moving from Melbourne City to Bangkok or vice versa can move to a place that is not too foreign to them in terms of the venues that is available within 1km of these places. By analyzing the venues using Foursquare and frequency of venue categories of places in Melbourne City and Bangkok we were able to get the top places in Melbourne City and Bangkok. By clustering these data we were able to get places in Melbourne City and Bangkok that are similar to each other in terms of the type of venues near to these places.

Due to lack of places in Melbourne City that are similar to places in Bangkok. We also analyzed places in Melbourne City that are similar to Melbourne, Melbourne City so that places in Melbourne City that are relatively similar to places in Bangkok can also be discovered.